share/eds/ods.go

```diff
@@ -0,0 +1,104 @@
+package eds
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+
+	cbor "github.com/ipfs/go-ipld-cbor"
+	"github.com/ipld/go-car"
+	"github.com/ipld/go-car/util"
+)
+
+var errNilReader = errors.New("ods-reader: can't create ODSReader over nil reader")
```

Comment:
Initially I've extracted as I usually do so it is not recreated on every occurrence. But now I've realised this is super rare event (it could be panic level of error) and doesn't need to be extracted.

Fixed spelling and moved removed `var`.